src/data_collectors/vae_data_collector.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import torch
+from torch import Tensor
+from torch.utils.data import TensorDataset
+
+from ..utils.step_record import RecordKeys as RK
+from .data_collector import DataCollector
+
+
+class VAEDataCollector(DataCollector):
```

Comment:
VAEのために作られたクラスですが、クラス名はその機能を簡潔に表す名称にします。
```suggestion
class ObservationCollector(DataCollector):
```